benchmark/core/Cabana_CommPerformance.cpp

```diff
@@ -348,20 +371,108 @@ void performanceTest( std::ostream& stream, const std::size_t num_particle,
                 comm_memory_space(), particles );
 
             s0 = Cabana::slice<0>( comm_particles );
-            Cabana::scatter( halo_fast, s0 );
+            Cabana::scatter( halo, s0 );
 
             s1 = Cabana::slice<1>( comm_particles );
-            Cabana::scatter( halo_fast, s1 );
+            Cabana::scatter( halo, s1 );
 
             s2 = Cabana::slice<2>( comm_particles );
-            Cabana::scatter( halo_fast, s2 );
+            Cabana::scatter( halo, s2 );
 
             s3 = Cabana::slice<3>( comm_particles );
-            Cabana::scatter( halo_fast, s3 );
+            Cabana::scatter( halo, s3 );
 
-            Cabana::deep_copy( particles, comm_particles );
+            particles = Cabana::create_mirror_view_and_copy(
+                data_memory_space(), comm_particles );
             halo_slice_scatter.stop( fraction );
         }
+
+        // Preallocated buffers.
+        // ---
+        double overallocation = 1.1;
+
+        // Run preallocated AoSoA buffer tests and time the ensemble.
+        auto comm_particles = Cabana::create_mirror_view_and_copy(
+            comm_memory_space(), particles );
+        auto gather = createGather( halo, comm_particles, overallocation );
```

Comment:
Now that I am seeing this, should the gather/scatter objects just store the aosoa/slice they are working on? The pattern we are going for here is that every aosoa/slice gets its own gather/scatter object to manage the buffer allocation. Then the apply call would just be `gather.apply()`

Reply:
Done. This is better